MDANSE/Src/MDANSE/Framework/Configurators/GroupingLevelConfigurator.py

```diff
@@ -72,18 +81,59 @@ def configure(self, value):
 
         SingleChoiceConfigurator.configure(self, value)
 
-        if value == "atom":
-            return
+        self["level"] = value
+
         trajConfig = self._configurable[self._dependencies["trajectory"]]
         atomSelectionConfig = self._configurable[self._dependencies["atom_selection"]]
         chemical_system = trajConfig["instance"].chemical_system
+
+        if value == "atom" or value == "each atom":
+            return
+
         indices = []
         elements = []
         names = []
         masses = []
+        group_names = []
+        group_elements = {}
+        group_n_atms = {}
         mass_lookup = chemical_system.atom_property("atomic_weight")
 
         if value == "molecule":
+            if len(trajConfig["instance"].chemical_system.unique_molecules()) == 0:
+                self.error_status = "The trajectory does not contain molecules."
+                return
+
+            for mol_name in chemical_system._clusters.keys():
+                n_atms = 0
+                group_name_elements = []
+                mol_selected = False
+                for mol_number, cluster in enumerate(
+                    chemical_system._clusters[mol_name]
+                ):
+                    for x in cluster:
+                        if x not in atomSelectionConfig["flatten_indices"]:
+                            continue
+                        mol_selected = True
+                        indices.append([x])
+                        elements.append([chemical_system.atom_list[x]])
+                        group_name_elements.append(chemical_system.atom_list[x])
+                        names.append(f"[{mol_name}]_{chemical_system.atom_list[x]}")
+                        masses.append([mass_lookup[x]])
+                        n_atms += 1
+                if mol_selected:
+                    group_names.append(mol_name)
+                    group_elements[mol_name] = group_name_elements
+                    group_n_atms[mol_name] = n_atms
+
+            self["name_to_element"] = {}
+            for name, element in zip(names, elements):
+                self["name_to_element"][name] = element[0]
+            self["group_names"] = sorted(set(group_names))
+            self["group_elements"] = group_elements
+            self["group_n_atms"] = group_n_atms
+
+        if value == "each molecule":
```

Comment:
If they're mutually exclusive:
```suggestion
        elif value == "each molecule":
```